src/types/collectors.rs

```diff
@@ -0,0 +1,155 @@
+/*
+    Copyright © 2023, ParallelChain Lab
+    Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+//! Defines the generic [`SignedMessage`] and [`Collector`] traits. 
+//! 
+//! Implementations used by the [Pacemaker][crate::pacemaker::types] and 
+//! [HotStuff][crate::hotstuff::types] protocols can be found in the respective modules. [`Collectors`]
+//! groups collectors for all active validator sets into a single struct, which can be easily updated
+//! on view or validator set state updates.
+
+pub use ed25519_dalek::{SigningKey, VerifyingKey, Signature};
+pub use sha2::Sha256 as CryptoHasher;
+
+use crate::messages::SignedMessage;
+use crate::state::{
+    block_tree::{BlockTree, BlockTreeError}, 
+    kv_store::KVStore
+};
+
+use super::{
+    basic::{ChainID, TotalPower, ViewNumber}, 
+    validators::{ValidatorSet, ValidatorSetState}
+};
+
+/// Evidence that a quorum of validators from a given validator set supports a given decision. The 
+/// evidence comes in the form of a set of signatures of the validators.
+pub trait Certificate {
+
+    /// Returns whether the certificate is correct, i.e., can serve as evidence that a given block is 
```

Comment:
This itemdoc says "i.e., can serve as evidence that a give **block** is approved...". This seems to only be applicable to `QuorumCertificate`s, not `TimeoutCertificate`s.

Reply:
Good catch, thanks